8.7.1 Release notes (#3137)
elastic/security-docs: docs/release-notes/8.7.asciidoc

[[release-notes-header-8.7.0]]
== 8.7

[discrete]
[[release-notes-8.7.1]]
=== 8.7.1

[discrete]
[[enhancements-8.7.1]]
==== Enhancements
There are no user-facing changes in 8.7.1.

[discrete]
[[bug-fixes-8.7.1]]
==== Bug fixes
* Fixes a UI bug that affected the rule details page when rules had long queries ({pull}153338[#153338]).

> * Fixes a bug that stopped Elastic Defend from working when Malware protection was enabled on Linux hosts.

Reviewer: ```suggestion
* Fixes a bug that stopped {elastic-defend} from working when malware protection was enabled on Linux hosts.
```

[[release-notes-header-8.7.0]]
== 8.7

[discrete]
[[release-notes-8.7.1]]
=== 8.7.1

[discrete]
[[enhancements-8.7.1]]
==== Enhancements
There are no user-facing changes in 8.7.1.

[discrete]
[[bug-fixes-8.7.1]]
==== Bug fixes

> * Fixes a bug that caused the {agent} upgrade from 8.6.2 to 8.7.0 to fail for {agents} running the Osquery Manager integration (https://github.com/elastic/elastic-agent/pull/2448[#2448]).

Reviewer 2: we should note here that upgrade `8.7.0` -> `8.7.1` could still cause an upgrade issue.  We should include the same workarounds as below.  After the upgrade to `8.7.1`, the problem will be fixed.

I'm referring to this section from 8.7.0 known issues.

```
The {agent} upgrade from 8.6.2 to 8.7.0 might fail for {agents} running the Osquery Manager integration (https://github.com/elastic/elastic-agent/issues/2433[#2433]). To prevent this, delete the Osquery Manager integration from your {agents} before upgrading them to 8.7.0, then re-add it after upgrading
```

[[release-notes-header-8.7.0]]
== 8.7

[discrete]
[[release-notes-8.7.1]]
=== 8.7.1

[discrete]
[[enhancements-8.7.1]]
==== Enhancements
There are no user-facing changes in 8.7.1.

[discrete]
[[bug-fixes-8.7.1]]
==== Bug fixes
* Fixes a bug that caused the {agent} upgrade from 8.6.2 to 8.7.0 to fail for {agents} running the Osquery Manager integration (https://github.com/elastic/elastic-agent/pull/2448[#2448]).

+

[NOTE]

=====

> If you are upgrading {agent} to 8.7.1 from 8.6.2 to 8.7.0, the {agent} upgrade might fail for {agents} running the Osquery Manager integration. To prevent this, delete the Osquery Manager integration from your {agents} before upgrading them to 8.7.0, then re-add it after upgrading. 

Reviewer: This seems to contradict the previous item, which says that we already fixed this bug. So why is the same behavior still happening? Is the idea that we _did_ fix the bug, but users might still have to delete and re-add the integration to get the fix? If that's the case, maybe rephrase this so it doesn't sound identical to the bug fix above. 

Also, should the last sentence be "before upgrading them to 8.7.1" since this is the 8.7.1 section?

Author: @joepeeples @kevinlog does the following revision work? It emphasizes that this is only a problem for users on 8.6.2 or 8.7.0, but that it's preventable.

```suggestion
To prevent upgrade issues while upgrading {agent} from 8.6.2 or 8.7.0, delete the Osquery Manager integration from your {agents} before upgrading them, then re-add it after upgrading. 

```

Reviewer: LGTM, thanks!

Reviewer 2: @nastasha-solomon this looks good to me, thanks!

=====

* Fixes a UI bug that affected the rule details page when rules had long queries ({pull}153338[#153338]).
* Fixes exception operator logic for mapping field conflicts ({pull}155071[#155071]).
* Fixes a bug that stopped {elastic-defend} from working when malware protection was enabled on Linux hosts.
* Fixes a bug that prevented policy artifacts from being properly updated after being added to the `.fleet-artifacts` index ({pull}154810[#154810]).
+

[NOTE]

=====
If the error message `Failed to download or validate user artifacts` appears in the Policy Response section of your Elastic Defend integration policy, you can resolve the error by adding the affected endpoint policy artifacts as <<trusted-apps-ov,global trusted applications>>, <<endpoint-rule-exceptions,endpoint exceptions>>, <<event-filters,event filters>>, <<blocklist,blocklist entries>>, or <<host-isolation-exceptions,host isolation exceptions>>. 

To find the affected endpoint policy artifacts:

. Go to *Fleet -> Agent policies*.
. Open the impacted {agent} policy.
. Select *Actions -> View policy*.

> . Search for `compression_algorithm: none` in the policy details. This field and value pair will appear under the affected endpoint policy artifacts.

Reviewer: ```suggestion
. Search for `compression_algorithm: none` in the policy details. This field-value pair will appear under the affected endpoint policy artifacts.
```

[[release-notes-header-8.7.0]]
== 8.7

[discrete]
[[release-notes-8.7.1]]
=== 8.7.1

[discrete]
[[enhancements-8.7.1]]
==== Enhancements
There are no user-facing changes in 8.7.1.

[discrete]
[[bug-fixes-8.7.1]]
==== Bug fixes
* Fixes a bug that caused the {agent} upgrade from 8.6.2 to 8.7.0 to fail for {agents} running the Osquery Manager integration (https://github.com/elastic/elastic-agent/pull/2448[#2448]).

+

[NOTE]

=====
To prevent upgrade issues while upgrading {agent} from 8.6.2 or 8.7.0, delete the Osquery Manager integration from your {agents} before upgrading them, then re-add it after upgrading. 

> If you already upgraded your {agents}, and they're are stuck in the `Updating` state, you'll need to upgrade them using the API instead of {fleet}. First, remove the Osquery integration from the {agents}, then use the https://petstore.swagger.io/?url=https://raw.githubusercontent.com/elastic/kibana/8.7/x-pack/plugins/fleet/common/openapi/bundled.json#/default/upgrade-agent[Agent upgrade API] to start the upgrade. After the upgrade completes, add the Osquery Manager integration to the {agents} again.

Author: Yes, I believe that's correct. Users will encounter the Agent upgrade issue only if they're upgrading from 8.6.2 or 8.7.0 to a later version.

cc: @kevinlog

> If you already upgraded your {agents}, and they're are stuck in the `Updating` state, you'll need to upgrade them using the API instead of {fleet}. First, remove the Osquery integration from the {agents}, then use the https://petstore.swagger.io/?url=https://raw.githubusercontent.com/elastic/kibana/8.7/x-pack/plugins/fleet/common/openapi/bundled.json#/default/upgrade-agent[Agent upgrade API] to start the upgrade. After the upgrade completes, add the Osquery Manager integration to the {agents} again.

Reviewer 2: @nastasha-solomon 

> Yes, I believe that's correct. Users will encounter the Agent upgrade issue only if they're upgrading from 8.6.2 or 8.7.0 to a later version.

This is correct. cc @jmikell821
